pinot-core/src/main/java/org/apache/pinot/core/util/GroupByUtils.java

```diff
@@ -36,6 +36,15 @@ public static int getTableCapacity(int limit) {
     return Math.max(limit * 5, NUM_RESULTS_LOWER_LIMIT);
   }
 
+  /**
+   * Returns the capacity of the table required by the given query.
+   * NOTE: It returns {@code max(limit * 5, resultLowerLimit)} where resultLowerLimit is configured by the user
+   *      (Default: 5000)
+   */
+  public static int getTableCapacity(int limit, int resultLowerLimit) {
```

Comment:
```suggestion
  public static int getTableCapacity(int limit, int minNumGroups) {
```

Reply:
Also change `NUM_RESULTS_LOWER_LIMIT` to public `DEFAULT_MIN_NUM_GROUPS` which can be accessed by the plan maker